tests/test_graph_utils.py

```diff
@@ -507,3 +509,103 @@ def test_result_equality(
 
         assert all(
             repeat(next(first_factory) == next(second_factory), nb_steps))  # type: ignore
+
+
+class TestSubgraphSampling:
+
+    @staticmethod
+    def get_graph(rng):
+        """
+        Randomly create somewhat complex graphs, then return a complex one
+        """
+        factory = GraphRandomFactory(
```

Comment:
I was thinking the arguments to GraphRandomFactory should be test parameters.
Maybe use a fixture, or at least define it somewhere else, outside the method.